### PR TITLE
CharmHub - Info

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following introduces queries to the charmhub store API. Now that all
requests go through a juju controller it makes it very trivial for other
clients like pylibjuju to gain the same powers as the go CLI.

The code is rather simple, it farms off a new type to encapsulate the
charmhub related calls (info and find which will follow) as the model
type is large and unwieldly, once there we just create a new facade and
perform the query.

Offically it would be nice to have to channel validation, but for now
we'll let the controller handle that.

### CODE REVIEW

File: juju/charmhub.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+from .errors import JujuError
+
+
+class CharmHub:
+    def __init__(self, model):
+        self.model = model
+
+    async def info(self, name, channel=None):
+        """info displays detailed information about a CharmHub charm. The charm
+        can be specified by name or by path.
```
Comment (hmlanigan) [inline juju/charmhub.py:11]: What does path mean in this context? The first thing I think of is a local charm

File: juju/charmhub.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+        """
+        if not name:
+            raise JujuError("name expected")
+
+        if channel is None:
+            channel = "latest"
```
Comment (hmlanigan) [inline juju/charmhub.py:18]: A channel cannot be a track only.  It'll fail at some point on the controller.  I'd prefer leaving channel blank if unspecified.  The caller won't know which channels are available.

File: juju/model.py [hunk 3]
```diff
@@ -776,6 +777,14 @@ def relations(self):
         """
         return list(self.state.relations.values())
 
+    @property
+    def charmhub(self):
+        """Return a charmhub store for requesting charm information using
```
Comment (hmlanigan) [inline juju/model.py:782]: nit: use of the word "store".  It's overused within juju land.  Repository?

File: examples/charmhub.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+"""
+Example to show how to connect to the current model and query the charm-hub
+store for information about a given charm.
```
Comment (hmlanigan) [inline examples/charmhub.py:3]: nit: use of the word "store". It's overused within juju land. Repository?

**Review by hmlanigan — COMMENTED** [review]
Reviewed from a charmhub perspective.

File: juju/charmhub.py (hunk 2 above)
Comment (SimonRichardson) [reply juju/charmhub.py:18]: Yes it can, channels are normalised from `latest/stable` to `latest`...

File: juju/charmhub.py (hunk 1 above)
Comment (SimonRichardson) [reply juju/charmhub.py:11]: I robbed it from Juju, I'll fix

**Review by petevg — APPROVED** [review]
Nice and simple. +1, LGTM.

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**